pkg/service/node.go

```diff
@@ -270,13 +285,18 @@ func (n *NodeService) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 		return nil, err
 	}
 
-	klog.Infof("Unmounting %s", req.GetTargetPath())
-	err := n.fsMounter.Unmount(req.GetTargetPath())
+	targetPath := req.GetTargetPath()
+	klog.Infof("Unmounting %s", targetPath)
+	err := n.mounter.Unmount(targetPath)
 	if err != nil {
 		klog.Infof("Failed to unmount")
 		return nil, err
 	}
 
+	if err = os.RemoveAll(targetPath); err != nil {
+		return nil, fmt.Errorf("remove target path: %w", err)
+	}
+
```

Comment:
related to above comments, if we don't create targetPath then we don't have to remove it

Reply:
Right but I do have to create the file so I can bind mount the block device on top of it (needs mount point).

Reply:
were we removing before?

Reply:
Nope, I believe that was a bug. We were leaving directories behind.